knative-operator/pkg/common/openshift.go

```diff
@@ -71,6 +74,61 @@ func ingress(ks *servingv1alpha1.KnativeServing, c client.Client) error {
 	return nil
 }
 
+// Update updates Knative controller env to use cluster wide proxy information
+func Update(ks *servingv1alpha1.KnativeServing, c client.Client) error {
+	var proxyEnv = map[string]string{
+		"HTTP_PROXY": os.Getenv("HTTP_PROXY"),
+		"NO_PROXY":   os.Getenv("NO_PROXY"),
+	}
+	deploy := &appsv1.Deployment{}
+	if err := c.Get(context.TODO(), client.ObjectKey{Name: "controller", Namespace: ks.GetNamespace()}, deploy); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	for c := range deploy.Spec.Template.Spec.Containers {
+		for k, v := range proxyEnv {
+			// If value is not empty then update deployment controller with env
+			if v != "" {
+				deploy.Spec.Template.Spec.Containers[c].Env = appendUnique(deploy.Spec.Template.Spec.Containers[c].Env, k, v)
+			} else {
+				// If value is empty then remove those keys from deployment controller
+				deploy.Spec.Template.Spec.Containers[c].Env = remove(deploy.Spec.Template.Spec.Containers[c].Env, k)
+			}
+		}
+	}
+	return c.Update(context.TODO(), deploy)
+}
+
+func remove(env []v1.EnvVar, key string) []v1.EnvVar {
+	for i := range env {
+		if env[i].Name == key {
+			return append(env[:i], env[i+1:]...)
+		}
+	}
+	return env
+}
+
+func appendUnique(orgEnv []v1.EnvVar, key, value string) []v1.EnvVar {
+	for i := range orgEnv {
+		if orgEnv[i].Name == key {
+			if value == "" {
```

Comment:
I don't quite parse this piece of code 🤔. When would this be empty and why'd remove things only to set their value afterwards?

I think this can be simplified to:

```golang
// Set the value if the key is already present.
for i := range orgEnv {
  if orgEnv[i].Name == key {
    orgEnv[i].Value = value
    return orgEnv
  }
}
// If not, append a key-value pair.
return append(orgEnv, v1.EnvVar{
  Name:  key,
	Value: value,
})
```